app/models/manageiq/providers/amazon/inventory/parser/cloud_manager.rb

```diff
@@ -164,50 +152,59 @@ def get_hardware_disks(instance, flavor)
     process_inventory_collection(disks, :disks) { |x| x }
   end
 
-  def parse_flavor(flavor)
-    name = uid = flavor[:name]
-
-    {
-      :type                     => ManageIQ::Providers::Amazon::CloudManager::Flavor.name,
-      :ext_management_system    => ems,
-      :ems_ref                  => uid,
-      :name                     => name,
-      :description              => flavor[:description],
-      :enabled                  => !flavor[:disabled],
-      :cpus                     => flavor[:vcpu],
-      :cpu_cores                => 1,
-      :memory                   => flavor[:memory],
-      :supports_32_bit          => flavor[:architecture].include?(:i386),
-      :supports_64_bit          => flavor[:architecture].include?(:x86_64),
-      :supports_hvm             => flavor[:virtualization_type].include?(:hvm),
-      :supports_paravirtual     => flavor[:virtualization_type].include?(:paravirtual),
-      :block_storage_based_only => flavor[:ebs_only],
-      :cloud_subnet_required    => flavor[:vpc_only],
-      :ephemeral_disk_size      => flavor[:instance_store_size],
-      :ephemeral_disk_count     => flavor[:instance_store_volumes]
-    }
+  def flavors
+    collector.flavors.each do |flavor|
+      name = uid = flavor[:name]
+      persister_flavor = persister.flavors.find_or_build(uid)
+
+      persister_flavor.assign_attributes(
+        :type                     => ManageIQ::Providers::Amazon::CloudManager::Flavor.name,
+        :ext_management_system    => ems,
+        :ems_ref                  => uid,
+        :name                     => name,
+        :description              => flavor[:description],
+        :enabled                  => !flavor[:disabled],
+        :cpus                     => flavor[:vcpu],
+        :cpu_cores                => 1,
+        :memory                   => flavor[:memory],
+        :supports_32_bit          => flavor[:architecture].include?(:i386),
```

Comment:
just to make sure, is `flavor[:architecture]` always an array of symbols?
`Array(flavor[:architecture]).map(&:to_sym)` would enforce it otherwise...